### PR TITLE
Avoid port conflicts in AddressRegistrationTests (#1721)

### PR DESCRIPTION
**Pull request by cesarbs**
#1721

There's still some room for conflicts in the \*`StaticPort`\* tests after 10 retries, but we have to test binding to a static port.

### CODE REVIEW

**Comment by natemcmaster** [comment]
LGTM. I assume it passes? It looks like we need to add rel/* to our travis/appveyor config so tests run on these branches.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs [hunk 1]
```diff
@@ -137,15 +181,51 @@ string FixUrl(string url)
...
-        private async Task RegisterIPEndPoint_Success(IPEndPoint endPoint, Func<IPEndPoint, string> testUrl)
+        private Task RegisterAddresses_Success(string addressInput, string testUrl, bool appendPortToTestUrl)
+            => RegisterAddresses_Success(addressInput, new[] { testUrl }, appendPortToTestUrl);
+
+        private async Task RegisterAddresses_StaticPort_Success(string addressInput, string[] testUrls)
+        {
+            var retryCount = 0;
+            var errors = new List<Exception>();
+
+            while (retryCount < MaxRetries)
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs]: How about less retries and a Delay?

Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs]: I thought of a delay too, but @halter73 pointed out that we're getting a new port each time, so why delay?

Comment (davidfowl) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs]: I see

**Pull request merged**